enhancements/cluster-logging/multi-cluster-log-forwarder.md

```diff
@@ -0,0 +1,246 @@
+
+This workflow supports any ClusterLogForwarder not created in the openshift-logging namespace and not named "instance".  The resource with this name in this namespace is significant to supporting the legacy workflow.
+
+**cluster administrator** is a user:
+
+* responsible for maintaining the cluster
+* able to bind cluster roles to serviceaccounts  
+* that deploys the **cluster-logging-operator**
+
+**namespace administrator** is a user:
+
+* able to create a serviceaccount
+* manages a **ClusterLogForwarder** custome resource
+
+The general worklow:
+
+* The namespace administrator creates a service account to be used by a log collector.  The service account must additionally include a token if there is intent to write to log storage managed by the Red Hat OpenShift logging subsystem.
+* The cluster administrator binds cluster roles to the service account for the log types they are allowed to collect (e.g. audit, infrastructure).  Several roles are added to the operator manifest and look something like:
+
+```
+    apiVersion: rbac.authorization.k8s.io/v1
+    kind: ClusterRole
+    metadata:
+      name: collect-audit-logs
+    rules:
+    - apiGroups:
+      - "logging.openshift.io"
+      resources:
+      - audit-logs
+      verbs:
+      - collect
+
+```
+This role allows collection of application logs and requires the namespace administor to bind the service account to the role like:
+
+```
+    oc create clusterrolebinding kube-audit-log-collection --clusterrole=collect-audit-logs --serviceaccount=openshift-kube-apiserver:audit-collector
+```
+
+* The namespace administrator creates a **ClusterLogForwarder** CR with a name that matches the service account and the inputs for which they are allowed to collect
+
+```yaml
+    apiVersion: "logging.openshift.io/v1"
+    kind: ClusterLogForwarder
+    metadata:
+      name: audit-collector
+      namespace: openshift-kube-apiserver
+    spec:
+      pipelines:
+       - inputRefs:
+         - audit
+         outputRefs:
+         - loki
+      outputs:
+      - name: loki
+        type: loki
+        url: https://mycenteralizedserver.some.place
+```
+
+* The namespace administrator creates a **ClusterLogging** CR with a name that matches the service account that only includes a spec for collection
+
+```yaml
+    apiVersion: "logging.openshift.io/v1"
+    kind: "ClusterLogging"
+    metadata:
+      name: audit-collector
+      namespace: openshift-kube-apiserver
+    spec:
+      collection:
+        type: "vector"
+```
+
+##### Verification and Validations 
+The operator will validate resources upon reconciliation of a **ClusterLogForwarder** and **ClusterLogging** CR.  Failure to meet any of the following conditions will stop the operator from deploying a collector and it will add error status to the resource.
+
+* The **ClusterLogForwarder** CR defines a valid spec
+* There exists a serviceaccount that is named the same as the **ClusterLogForwarder** CR
```

Comment:
I like the same-name convention for associating our own resources (CLF, CL, future additional resources) but specifically for service accounts: perhaps we should follow the same spec.serviceAccount convention that is used for Pods, since this is already familiar to users. I'm not dead set either way, but seems like there's a case for being consistent with pre-existing k8s convention here.

Reply:
@alanconway My original suggestion was for the spec to tell us the SA to use and we went away from that for reasons I do not recall.  I think by moving to this pattern it will also resolve my other concerns regarding how we keep the "legacy" workflow from being a unicorn from the mutli-cluster workflow.  By specifying the SA, we should be able to migrate the legacy to new naming patterns and only need to require or rely upon only a specially named SA to support things like cloudwatch upgrade.  What say you?.... `spec.serviceAccount` or name convention?

Reply:
Prefer specifying the SA name instead of a magic value.